test/snippet/index/bi_fm_index_iterator.cpp

```diff
@@ -0,0 +1,92 @@
+#include <vector>
+#include <iostream>
+#include <seqan3/index/all.hpp>
+
+using namespace seqan3;
+using namespace seqan3::literal;
+
+int main()
+{
+    {
+        //! [cycle]
+        std::vector<dna4> genome{"GAATTAATGAAC"_dna4};
+        bi_fm_index<std::vector<dna4>> index{genome}; // build the index
+
+        // TODO(comments): outputs [A, A, G] instead of AAG
+
+        auto it = index.begin(); // create an iterator
+        // it.cycle_back(); // cycle_back / cycle_front on begin() is undefined behaviour!
+        it.extend_right("AAC"_dna4); // search the sequence "AAC"
+        std::cout << it.query() << '\n'; // outputs "AAC"
+        std::cout << it.last_char() << '\n'; // outputs 'C'
+
+        // it.cycle_front(); // undefined behaviour! only cycle_back() is allowed after extend_right()
+        it.cycle_back(); // search the sequence "AAT"
+        std::cout << it.query() << '\n'; // outputs "AAT"
+        std::cout << it.last_char() << '\n'; // outputs 'T'
+
+        it.extend_left(dna4::G); // search the sequence "GAAT"
+        std::cout << it.query() << '\n'; // outputs "GAAC"
+        std::cout << it.last_char() << '\n'; // outputs 'G'
+
+        // it.cycle_back(); // undefined behaviour! only cycle_front() is allowed after extend_left()
+        it.cycle_front(); // search the sequence "TAAT"
+        std::cout << it.query() << '\n'; // outputs "TAAT"
+        std::cout << it.last_char() << '\n'; // outputs 'T'
+
+        it.cycle_front(); // search the sequence "TAAT"
+        std::cout << it.query() << '\n'; // outputs "TAAT"
+        std::cout << it.last_char() << '\n'; // outputs 'T'
+        //! [cycle]
+    }
+
+    std::cout << "-------------------------\n";
```

Comment:
why are you printing this?

Reply:
In case a user wants to compile the snippet it's hard to see where the output of the different blocks start and end. But I'll make a more meaningful output instead of ------ :smiley: